packages/create-rspack/index.js

```diff
@@ -4,34 +4,51 @@ const { hideBin } = require("yargs/helpers");
 const fs = require("fs");
 const path = require("path");
 const prompts = require("prompts");
+const { formatTargetDir } = require("./utils");
 yargs(hideBin(process.argv))
-	.command("$0", "init rspack project", argv => {
+	.command("$0", "init rspack project", async argv => {
 		const { help } = argv.argv;
 		if (help) return;
 		const defaultProjectName = "rspack-project";
-		prompts([
+		let targetDir = defaultProjectName;
+		await prompts([
 			{
 				type: "text",
 				name: "projectDir",
 				initial: defaultProjectName,
-				message: "Project folder"
+				message: "Project folder",
+				onState: state => {
+					targetDir = formatTargetDir(state.value) || defaultProjectName;
+				}
 			}
-		]).then(result => {
-			const { projectDir } = result;
-			const root = path.resolve(process.cwd(), projectDir);
-			if (fs.existsSync(root)) {
-				throw new Error("project directory already exists");
-			}
-			fs.mkdirSync(root);
-			// TODO support more template in the future
-			const templateDir = "template-react";
-			const srcFolder = path.resolve(__dirname, templateDir);
-			copyFolder(srcFolder, projectDir);
-			console.log("\nDone. Now run:\n");
-			console.log(`cd ${projectDir}\n`);
-			console.log(`npm install\n`);
-			console.log(`npm run dev\n`);
-		});
+		]);
+		let root = path.resolve(process.cwd(), targetDir);
+		while (fs.existsSync(root)) {
+			console.log(
+				`${targetDir} is not empty, please choose another project name`
+			);
+			await prompts([
+				{
+					type: "text",
+					name: "projectDir",
+					initial: defaultProjectName,
+					message: "Project folder",
+					onState: state => {
+						targetDir = formatTargetDir(state.value) || defaultProjectName;
+					}
+				}
+			]);
+			root = path.resolve(process.cwd(), targetDir);
+		}
+		fs.mkdirSync(root, { recursive: true });
+		// TODO support more template in the future
+		const templateDir = "template-react";
+		const srcFolder = path.resolve(__dirname, templateDir);
+		copyFolder(srcFolder, targetDir);
+		console.log("\nDone. Now run:\n");
+		console.log(`cd ${targetDir}\n`);
+		console.log(`npm install\n`);
```

Comment:
Don't miss this: https://github.com/web-infra-dev/rspack/pull/2146

Reply:
OK,I will change it soon

Reply:
Would you reset the whole commit and force push it ? The other contributor failed to sign-up the CLA

Reply:
OK, I will